tests/phpunit/includes/testcase-ajax.php

```diff
@@ -154,17 +154,11 @@ public function setUp() {
 	/**
 	 * Tears down the test fixture.
 	 *
-	 * Resets $_POST, removes the wp_die() override, restores error reporting.
+	 * Resets global state, restores error reporting.
 	 */
 	public function tearDown() {
-		$_POST = array();
-		$_GET  = array();
```

Comment:
This is completely correct, but I think it's safe to assume that a `setUp()` method must always call its parent `setUp()` method (same for `tearDown()`) otherwise all the assumptions that are made about the state of the test suite before a test runs are invalid.

It would be great if there was a way of causing a test to fail if it doesn't call the `setUp()` method, but off the top of my head I can't think of a way to do this.

Reply:
> otherwise all the assumptions that are made about the state of the test suite before a test runs are invalid.

Except that those assumptions with the current `setUp()`and `tearDown()` are invalid anyways.

`setUp()` should prepare the environment for the tests.
`tearDown()` should ensure that the environment is brought back to its default state.

The fact that the "bring back to its default state" is now, for a significant part, done in the `setUp()`, makes the test suite unstable by nature.

> It would be great if there was a way of causing a test to fail if it doesn't call the setUp() method, but off the top of my head I can't think of a way to do this.

Well, I could create a sniff for that, but that's outside the scope of this ticket.